File: RDMPEG/RDMPEGConverter/RDMobileFFmpegStatistics/RDMobileFFmpegStatistics.m
//
//  RDMobileFFmpegStatistics.m
//  RDMPEG
//
//  Created by Serhii Alpieiev on 28.08.2021.
//  Copyright © 2021 Readdle. All rights reserved.
//

#import "RDMobileFFmpegStatistics.h"
#import <FFmpegKit/FFmpegKit.h>



NS_ASSUME_NONNULL_BEGIN

@interface RDMobileFFmpegStatistics ()

@property (nonatomic, readonly) Statistics *statistics;

@end



@implementation RDMobileFFmpegStatistics

#pragma mark - Lifecycle

- (instancetype)initWithStatistics:(Statistics *)statistics {
    NSParameterAssert(statistics);
    
    self = [super init];
    
    if (nil == self) {
        return nil;
    }
    
    _statistics = statistics;
    
    return self;
}

#pragma mark - Accessors

- (NSInteger)frameNumber {
    return [self.statistics getVideoFrameNumber];
}

- (double)fps {
    return [self.statistics getVideoFps];
}

- (double)quality {
    return [self.statistics getVideoQuality];
}

- (NSInteger)size {
    return [self.statistics getSize];
}

- (NSInteger)time {
    return [self.statistics getTime];
}

- (double)bitrate {
    return [self.statistics getBitrate];
}

- (double)speed {
    return [self.statistics getSpeed];
}


@end

NS_ASSUME_NONNULL_END

File: RDMPEG/RDMPEGConverter/RDMobileFFmpegStatistics/RDMobileFFmpegStatistics.swift
//
//  RDMobileFFmpegStatistics.swift
//  RDMPEG
//
//  Created by Max on 16/07/2024.
//  Copyright © 2024 Readdle. All rights reserved.
//

import Foundation
import ffmpegkit

@objcMembers
public class RDMobileFFmpegStatistics: NSObject {
    private let statistics: Statistics

    public var frameNumber: Int {
        return Int(statistics.getVideoFrameNumber())
    }

    public var fps: Float {
        return statistics.getVideoFps()
    }

    public var quality: Float {
        return statistics.getVideoQuality()
    }

    public var size: Int {
        return statistics.getSize()
    }

    public var time: Int {
        return Int(statistics.getTime())
    }

    public var bitrate: Double {
        return statistics.getBitrate()
    }

    public var speed: Double {
        return statistics.getSpeed()
    }

    public init(statistics: Statistics) {
        self.statistics = statistics
    }
}
